src/yunohost/settings.py

```diff
@@ -39,6 +39,10 @@
     ("security.password.admin.strength", {"type": "int", "default": 1}),
     ("security.password.user.strength", {"type": "int", "default": 1}),
     ("service.ssh.allow_deprecated_dsa_hostkey", {"type": "bool", "default": False}),
+    ("service.ssh.ciphers.compatibility", {"type": "enum", "default": "modern",
+        "choices": ["intermediate", "modern"]}),
+    ("security.ciphers.compatibility", {"type": "enum", "default": "intermediate",
+        "choices": ["intermediate", "modern"]}),
```

Comment:
Oh and then the question which goes beyond this PR is : what do we want to do when those settings are modified ... So e.g. when calling `yunohost setting set security.ciphers.compatibility modern` (whatever the exact syntax), ssh or nginx conf shall be regenerated ... we can't expect the user to guess that.

So, when this setting system was first designed, we though of using a decorator for this to create something like : 

```
@when_changing("some_setting_name")
def some_function_name():
   [...]
```

so that for this case that would be something like : 

```
@when_changing("security.ciphers.compatibility")
def some_function_name():
    service_regen_conf("nginx")
```

So this `@when_changing` decorator needs to be defined somewhere so that after using `setting_set`, the functions decorated in that way are called :wink: 

Dunno if you are at ease with decorators and understand this whole thing, otherwise I can work on it at some point.

Reply:
I understand the mecanism but I'm not confortable enough with Yunohost stack to handle that at the moment.
If it's a new mecanism to design I bet you or someone with more experience on this subject is to be needed.

Reply:
So I implemented the mechanism in https://github.com/YunoHost/yunohost/pull/654

If the design does not change, it should be as simple as adding this kind of lines : 

```python
@post_change_hook("service.ssh.ciphers.compatibility")
def foo(old_value, new_value):
    # (We don't really care about the old and new value for that use case)
    service_regen_conf(["ssh"])
```

Reply:
As soon as I am able to test this mecanism in the context of this PR, I will report here. Thx @alexAubin for your help.

Reply:
I have updated this PR to use post_change_hook. It works like a charm ! 👍